types/ember-data/index.d.ts

```diff
@@ -25,7 +25,30 @@ import AdapterRegistry from 'ember-data/types/registries/adapter';
 type ModelKeys<Model extends DS.Model> = Exclude<keyof Model, keyof DS.Model>;
 
 type AttributesFor<Model extends DS.Model> = ModelKeys<Model>; // TODO: filter to attr properties only (TS 2.8)
-type RelationshipsFor<Model extends DS.Model> = ModelKeys<Model>; // TODO: filter to hasMany/belongsTo properties only (TS 2.8)
+
+type UnboxBelongsTo<T> =
+    T extends DS.AsyncBelongsTo<infer C> ? C
+        : T extends DS.Model ? T
+        : T extends DS.AsyncHasMany<any> ? never
+        : T extends DS.SyncHasMany<any> ? never
+        : DS.Model;
+
+type UnboxHasMany<T> =
+    T extends DS.AsyncHasMany<infer C> ? C
+      : T extends DS.SyncHasMany<infer C> ? C
+      : T extends DS.AsyncBelongsTo<any> ? never
+      : T extends DS.Model ? never
+      : DS.Model;
+
+type BelongsToModels<Model extends DS.Model> = {
+    [k in keyof Model]: UnboxBelongsTo<Model[k]>
```

Comment:
Purely a stylistic preference, but can we make `k` a capital letter?

 This helps to disambiguate between constrained and unconstrained index signatures (e.g., `{ [k: string]: number }` where `k` is not a type, and `{ [K in keyof Model]: string }` where `K` is a type)